service/src/main/java/bio/terra/profile/service/profile/flight/create/LinkBillingProfileIdToMrgStep.java

```diff
@@ -0,0 +1,51 @@
+package bio.terra.profile.service.profile.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.profile.service.azure.ApplicationService;
+import bio.terra.profile.service.profile.exception.MissingRequiredFieldsException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+
+/** Add a tag that associates a billing profile ID with an azure managed resource group */
+public record LinkBillingProfileIdToMrgStep(
+    ApplicationService applicationService, BillingProfile profile, AuthenticatedUserRequest user)
+    implements Step {
+  private static final String BILLING_PROFILE_ID_TAG = "terra.billingProfileId";
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    try {
+      if (profile.tenantId().isEmpty()) {
+        throw new MissingRequiredFieldsException(
+                "No tenant ID on billing profile ID " + profile.id());
+      }
```

Comment:
I think you may need to do more to fix the sonar error. One possible fix:
```suggestion
      var tenantId = profile.tenantId().orElseThrow(() -> new MissingRequiredFieldsException(
                "No tenant ID on billing profile ID " + profile.id());               
       ...
       applicationService.addTagToMrg(tenantId, ...);
```